value/src/it/functional/src/test/java/com/google/auto/value/GradleTest.java

```diff
@@ -0,0 +1,171 @@
+package com.google.auto.value;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GradleTest {
+  @Rule public TemporaryFolder fakeProject = new TemporaryFolder();
+
+  private static final String BUILD_GRADLE_TEXT =
+      String.join("\n",
+          "plugins {",
+          "  id 'java-library'",
+          "}",
+          "repositories {",
+          "  maven { url = uri('${localRepository}') }",
+          "}",
+          "dependencies {",
+          "  compileOnlyApi      'com.google.auto.value:auto-value-annotations:${autoValueVersion}'",
+          "  annotationProcessor 'com.google.auto.value:auto-value:${autoValueVersion}'",
+          "  testImplementation  'junit:junit:${junitVersion}'",
```

Comment:
Where is this value coming from? I don't see it anywhere in the repo. I suppose the build doesn't fail because the `test` task is skipped due to the absence of test sources, and possibly resolution of this dependency isn't even attempted as a result?

You could probably just remove that line, and possibly use `compileJava` or `classes` rather than `build` task in `buildFakeProject`.
```suggestion
```

Reply:
Thanks, that was left over from earlier experiments.